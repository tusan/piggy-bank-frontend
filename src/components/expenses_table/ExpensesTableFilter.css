.expenses-filter button.ant-btn.ant-btn-primary,
.expenses-filter .ant-row.ant-form-item,
.expenses-filter .ant-form-item-control-wrapper {
  width: 100%;
}

.expenses-filter form.ant-form.ant-form-inline {
  text-align: left;
}

.expenses-filter {
  border: 1px solid;
  border-color: #e8e8e8;
  margin-top: 20px;
  border-radius: 5px;
  padding: 10px;
}

@media screen and (min-width: 700px) {
  .expenses-filter button.ant-btn.ant-btn-primary,
  .expenses-filter .ant-row.ant-form-item {
    width: auto;
  }

  .expenses-filter .ant-row.ant-form-item {
    width: 30%;
  }
}
