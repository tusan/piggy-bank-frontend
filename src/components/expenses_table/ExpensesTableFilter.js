import React from 'react';
import { Form, Button, DatePicker } from 'antd';
import './ExpensesTableFilter.css';

const { Item } = Form;

class ExpensesTableFilter extends React.PureComponent {
  render = () => {
    const { getFieldDecorator } = this.props.form;
    return (
      <div className="expenses-filter">
        <Form layout="inline" onSubmit={this.handleSubmit}>
          <h3 className="form-title">Select Period</h3>

          <Item>
            {getFieldDecorator('dateStart', {
              initialValue: this.props.dateStartInit
            })(
              <DatePicker
                className="expense-date-picker"
                placeholder="Start Date"
                format={'DD/MM/YYYY'}
              />
            )}
          </Item>
          <Item>
            {getFieldDecorator('dateEnd', {
              initialValue: this.props.dateEndInit
            })(
              <DatePicker
                className="expense-date-picker"
                placeholder="End Date"
                format={'DD/MM/YYYY'}
              />
            )}
          </Item>
          <Item>
            <Button type="primary" htmlType="submit">
              Filter
            </Button>
          </Item>
        </Form>
      </div>
    );
  };

  handleSubmit = e => {
    e.preventDefault();

    this.props.form.validateFields((err, values) => {
      if (!err) {
        this.props.handleFilterChange(values);
      }
    });
  };
}
export default Form.create()(ExpensesTableFilter);
