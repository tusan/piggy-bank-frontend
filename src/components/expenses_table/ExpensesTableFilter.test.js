import React from 'react';
import renderer from 'react-test-renderer';
import ExpensesTableFilter from './ExpensesTableFilter';
import moment from 'moment';

describe('<ExpensesTableFilter />', () => {
  it('should render with default values', () => {
    const handleFilterChange = jest.fn();
    const dateStart = moment('20181126');
    const dateEnd = moment('20181226');

    const filterForm = renderer
      .create(
        <ExpensesTableFilter
          handleFilterChange={handleFilterChange}
          dateStartInit={dateStart}
          dateEndInit={dateEnd}
        />
      )
      .toJSON();
    expect(filterForm).toMatchSnapshot();
  });
});
