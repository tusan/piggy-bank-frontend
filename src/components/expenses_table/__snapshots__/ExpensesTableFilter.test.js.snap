// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExpensesTableFilter /> should render with default values 1`] = `
<div
  className="expenses-filter"
>
  <form
    className="ant-form ant-form-inline"
    onSubmit={[Function]}
  >
    <h3
      className="form-title"
    >
      Select Period
    </h3>
    <div
      className="ant-row ant-form-item"
    >
      <div
        className="ant-form-item-control-wrapper"
      >
        <div
          className="ant-form-item-control has-success"
        >
          <span
            className="ant-form-item-children"
          >
            <span
              className="expense-date-picker ant-calendar-picker"
              id="dateStart"
              onBlur={[Function]}
              onFocus={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={Object {}}
            >
              <div
                onClick={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onTouchStart={[Function]}
              >
                <input
                  className="ant-calendar-picker-input ant-input"
                  placeholder="Start Date"
                  readOnly={true}
                  value="26/11/2018"
                />
                <i
                  className="anticon anticon-close-circle ant-calendar-picker-clear"
                  onClick={[Function]}
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="close-circle"
                    fill="currentColor"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 0 1-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
                    />
                  </svg>
                </i>
                <i
                  className="anticon anticon-calendar ant-calendar-picker-icon"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="calendar"
                    fill="currentColor"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
                    />
                  </svg>
                </i>
              </div>
            </span>
          </span>
        </div>
      </div>
    </div>
    <div
      className="ant-row ant-form-item"
    >
      <div
        className="ant-form-item-control-wrapper"
      >
        <div
          className="ant-form-item-control has-success"
        >
          <span
            className="ant-form-item-children"
          >
            <span
              className="expense-date-picker ant-calendar-picker"
              id="dateEnd"
              onBlur={[Function]}
              onFocus={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={Object {}}
            >
              <div
                onClick={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onTouchStart={[Function]}
              >
                <input
                  className="ant-calendar-picker-input ant-input"
                  placeholder="End Date"
                  readOnly={true}
                  value="26/12/2018"
                />
                <i
                  className="anticon anticon-close-circle ant-calendar-picker-clear"
                  onClick={[Function]}
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="close-circle"
                    fill="currentColor"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 0 1-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
                    />
                  </svg>
                </i>
                <i
                  className="anticon anticon-calendar ant-calendar-picker-icon"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="calendar"
                    fill="currentColor"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
                    />
                  </svg>
                </i>
              </div>
            </span>
          </span>
        </div>
      </div>
    </div>
    <div
      className="ant-row ant-form-item"
    >
      <div
        className="ant-form-item-control-wrapper"
      >
        <div
          className="ant-form-item-control"
        >
          <span
            className="ant-form-item-children"
          >
            <button
              className="ant-btn ant-btn-primary"
              onClick={[Function]}
              type="submit"
            >
              <span>
                Filter
              </span>
            </button>
          </span>
        </div>
      </div>
    </div>
  </form>
</div>
`;
